### ISSUE TITLE
Send team invitation hard

### ISSUE THREAD

**Issue by hyamanieu** [FlowFuse/flowfuse#3086]
### Description

This is a related to #3039 but from the point of view of a team owner inviting users.

## The current status:

When I need to add people to a team, the only way is to:
- know exactly the username of the member(s) I want to add by asking them (if not an Administrator) or go over to the list of users (if an admin)
- go to the team area
- click on "Members" on the left hand side
- click on "invite member" on the top right hand corner
- write exactly the username of the member to invite
- click invite
- repeat previous step for each member

## What I would expect:

Two ways to send an invitation:

- as a FlowFurse administrator, from the list of users:
  - search through user
  - click the 3 dots **⋮**
  - have a "add to teams" button
  - be able to add them to all the teams I need
- as either a FlowFuse admin or a team owner, from the team area
  - click on Members on the left hand side
  - Have an "Invite Member**s**" button
  - type the first 3 letters then have a suggestion with a list of users based on: a username, a user e-mail address, a user full name.
  - click on the wished user or simply press tab if he/she is the first in the list
  - type a "coma" or semicolon as separator
  - repeat the two last steps for each new member
  - click "invite" to send all invitations to all selected users
  - (optionally: send separate invitation for people with other roles)

### Have you provided an initial effort estimate for this issue?

I am not a FlowForge team member

**Comment by joepavitt** [comment]
> as a FlowFuse administrator, from the list of users:

Very reasonable, and makes good sense.

> as either a FlowFuse admin or a team owner, from the team area
 
Slight concerns on this though with compromising data as a Team Owner. That would provide data on _every_ user available on the FlowFuse platform to _any_ team owner, even if 3-letters were required first, matching company names, common e-mail domains, or first names would be fairly straight forward. 

On FlowFuse Cloud for example, we have 4k+ users, all of which would then be searchable by any team owner, which feels a little _too_ much.

In the many invite-based tools I have seen, having to put in a full e-mail address is fairly standard practice? As a FF Admin though, I understand the value, and am not as concerned about the compromise.

**Comment by hyamanieu** [comment]
> > as either a FlowFuse admin or a team owner, from the team area
> 
> Slight concerns on this though with compromising data as a Team Owner. That would provide data on _every_ user available on the FlowFuse platform to _any_ team owner, even if 3-letters were required first, matching company names, common e-mail domains, or first names would be fairly straight forward.
> 
> On FlowFuse Cloud for example, we have 4k+ users, all of which would then be searchable by any team owner, which feels a little _too_ much.
> 
> In the many invite-based tools I have seen, having to put in a full e-mail address is fairly standard practice? As a FF Admin though, I understand the value, and am not as concerned about the compromise.

I was unaware of this user case. As far as I understand the prompt: I have to type in the **real** username, not the email. The email would be easier, I haven't tried as it's written the username is expected (and unfortunately the username doesn't have a regex policy like email addresses, so they're different from the email addresses after removing the domain name)

Perhaps rewording as such:

- instead of write the username of a single user and repeat for each new team member
- ==> be able to type a list of emails or user names before sending the invitation

**Comment by joepavitt** [comment]
> ==> be able to type a list of emails or user names before sending the invitation

Yep, I like it. Also agree that email should also be an option here. @MarianRaphael FYI
